container/kvm/doc.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+/*
+Package kvm provides the facilities to deploy to kvm instances.
+
+kvm implements the container interface for worker/provisioner to manage
+applications deployed to kvm based 'containers' (see
+juju/worker/provisioner/provisioner.go:containerProvisioner and
+juju/container/container.go:container.Manager). The worker provisioner
+specifics are in juju/worker/provisioner/kvm-broker.go and
+juju/worker/container_initilisation.go.
+
+The provisioner worker manages kvm containers through the interface provided in
+this package, see: containerfactory.go, container.go, instance, and
+initialization.go.  That is to say those files provide the container.Manager
+interface while the rest of this package are the implmentation of
+container.Manager for kvm instances.
+
+This package originally depended on the ubuntu uvtool apt package. This meant
+that kvm would only work on ubuntu on amd64. The goal of removing Juju's
+dependency on uvtool is to allow kvm to also work on arm64 and ppc64el.
+However, it is still only expected to work on ubuntu.
+
+When removing uvtool we (redir) performed a survey of the libvirt and qemu go
+package landscape. There are a number of cgo based libraries and two possibly
+promising ones once they are further developed:
+github.com/digitalocean/go-libvirt and github.com/digitalocean/go-qemu. Those
+packages are nascent and alpha at the time of this writing. They implement pure
+go interfaces to libvirt and qemu by way of libvirt/qemu's custom RPC protocol.
+While this would reduce the number of commnads that require shelling out, it
+wouldn't provide a way to create and manage disk images. So unless someone
+implements qemu-utils and genisoimage in go, we'll always need to shell out for
+those calls.  The wrapped commands exist, shockingly, in wrappedcmds.go with
+the exception of libvirt pool initialisation bits which are in
+initialization.go.
+
+After the provisioner initializes the kvm environment, we synchronise (fetch if
+we don't have one) an ubuntu qcow image for the appropriate series and
+architecture. This happens in sync.go and uses Juju's simplestreams
```

Comment:
I think I am having pronoun problems. Sync or fetching used to happen in uvtool and we shelled out to it. It depended on a python simplestreams impl. Juju already had some simplestreams parsing code for cloud images and tools (agents) streams. I added the bits for also handling content-download stream which is dependent on the existing simplestreams code. That is why the simplestreams code lives in juju/juju. The kvm code implements kvm as a juju container, so it is not useful outside juju/juju.